TriblerGUI/widgets/downloadspage.py

```diff
@@ -128,15 +128,11 @@ def on_received_downloads(self, downloads):
                 self.window().download_details_widget.update_pages()
 
         # Check whether there are download that should be removed
-        toremove = set()
-        for infohash, item in self.download_widgets.iteritems():
+        for infohash, item in self.download_widgets.copy().iteritems():
```

Comment:
You can just use `.items()` instead of `.copy().iteritems()`.

Reply:
Would you prefer to use .items() instead of .iteritems()? Since other part of this file is in python 2, I picked .iteritems()

.copy() is necessary. We delete items during the iteration, this is not allow in a for-loop.

Reply:
Yes I would prefer `.items()`, which is perfectly valid Python 2: https://docs.python.org/2/library/stdtypes.html#dict.items

Excerpt from the documentation:
> Return a copy of the dictionary’s list of (key, value) pairs.

Reply:
In python 3 it returns a new view instead of a copy of list, would that be safe to remove itemd from it?

Reply:
If we (ever) switch to Python 3, we can use `frozenset()` 👍 .